structured/db.go

```diff
@@ -19,74 +19,28 @@
 package structured
 
 import (
-	"bytes"
-	"encoding/gob"
-
 	"github.com/cockroachdb/cockroach/client"
-	"github.com/cockroachdb/cockroach/keys"
 	"github.com/cockroachdb/cockroach/proto"
 )
 
-// A DB interface provides methods to access a datastore
-// using a structured data API.
-type DB interface {
-	PutSchema(*Schema) error
-	DeleteSchema(*Schema) error
-	GetSchema(string) (*Schema, error)
-}
-
-// A structuredDB satisfies the DB interface using the
-// Cockroach kv client API.
-type structuredDB struct {
-	// kvDB is a client to the monolithic key-value map.
-	kvDB *client.DB
+// DB provides Cockroach's Structured Data API.
+type DB struct {
+	// db is a client to the monolithic key-value map.
+	db *client.DB
 }
 
 // NewDB returns a key-value datastore client which connects to the
 // Cockroach cluster via the supplied gossip instance.
-func NewDB(kvDB *client.DB) DB {
-	return &structuredDB{kvDB: kvDB}
+func NewDB(db *client.DB) *DB {
+	return &DB{db: db}
 }
 
-// PutSchema inserts s into the kv store for subsequent
-// usage by clients.
-func (db *structuredDB) PutSchema(s *Schema) error {
-	if err := s.Validate(); err != nil {
-		return err
-	}
-	k := keys.MakeKey(keys.SchemaPrefix, proto.Key(s.Key))
-	// TODO(pmattis): This is an inappropriate use of gob. Replace with
-	// something else.
-	var buf bytes.Buffer
-	if err := gob.NewEncoder(&buf).Encode(s); err != nil {
+func (s *DB) CreateTable(args *proto.CreateTableRequest, reply *proto.CreateTableResponse) error {
+	desc, err := s.db.CreateTableInner(args.Schema)
+	if err != nil {
 		return err
 	}
-	return db.kvDB.Put(k, buf.Bytes())
-}
+	reply.TableID = desc.ID
```

Comment:
As discussed in person, let's just drop `TableID` from the reply until we need it. This would get rid of the need for `CreateTableInner`.